inst/templates/ad_adpc.R

```diff
@@ -0,0 +1,513 @@
+
+# ---- Load source datasets ----
+
+# Use e.g. haven::read_sas to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+
+# Load PC, EX, VS and ADSL
+data("admiral_pc")
+data("admiral_ex")
+data("admiral_vs")
+
+data("admiral_adsl")
+
+adsl <- admiral_adsl
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values
+
+# Load EX
+
+ex <- convert_blanks_to_na(admiral_ex)
+
+# Load PC and fix format issue for PCDTC
+
+pc <- convert_blanks_to_na(admiral_pc) %>%
+  rename(PCDTM = PCDTC) %>%
+  mutate(PCDTC = as.character.Date(PCDTM, format = "%Y-%m-%dT%H:%M:%S")) %>%
+  select(-PCDTM)
+
+# Load VS for baseline height and weight
+
+vs <- convert_blanks_to_na(admiral_vs)
+
+# ---- Lookup tables ----
+param_lookup <- tibble::tribble(
+  ~PCTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "XAN", "XAN", "Pharmacokinetic concentration of Xanomeline", 1,
+  "DOSE", "DOSE", "Xanomeline Patch Dose", 2,
+)
+
+
+attr(param_lookup$PCTESTCD, "label") <- "Pharmacokinetic Test Short Name"
+
+# ---- User defined functions ----
+
+# Here are some examples of how you can create your own functions that
+# operates on vectors, which can be used in `mutate`.
+format_avalcat1n <- function(param, aval) {
+  case_when(
+    param == "PKCONC" & aval < 1 ~ 1,
+    param == "PKCONC" & aval >= 1 ~ 2,
+    T ~ NA_real_
+  )
+}
+
+
+# ---- Derivations ----
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- vars(TRTSDT, TRTSDTM, TRT01P, TRT01A)
+
+adpc <- pc %>%
+  # Join ADSL with PC (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+  # Calculate ADTM, ADT, ADY
+  derive_vars_dtm(
+    new_vars_prefix = "A",
+    dtc = PCDTC,
+    date_imputation = "FIRST",
+    time_imputation = "00:00:00",
+    ignore_seconds_flag = FALSE,
+    flag_imputation = "auto"
+  ) %>%
+  # Derive dates and times from date/times
+  derive_vars_dtm_to_dt(vars(ADTM)) %>%
+  derive_vars_dtm_to_tm(vars(ADTM)) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = vars(ADT)) %>%
+  # Derive event ID and nominal relative time from first dose (NFRLT)
+  mutate(
+    EVID = 0,
+    DRUG = PCTEST,
+    NFRLT = if_else(PCTPTNUM < 0, 0, PCTPTNUM), .after = USUBJID
+  )
+
+
+# ---- Get dosing information ----
+
+ex <- ex %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+  # Keep records with nonzero dose
+  filter(EXDOSE > 0) %>%
+  # Add time and set missing end date to start date
+  # Impute missing time to 00:00:00
+  # Derive Analysis Start and End Dates
```

Comment:
Can we add this note to t Lines 87-93 as well?

Reply:
Yes, updated comments to clarify.